contrib/epee/include/net/net_helper.h

```diff
@@ -487,79 +415,6 @@ namespace net_utils
 
 		}
 
-		inline bool recv_n(std::string& buff, int64_t sz, std::chrono::milliseconds timeout)
-		{
-
-			try
-			{
-				// Set a deadline for the asynchronous operation. Since this function uses
-				// a composed operation (async_read_until), the deadline applies to the
-				// entire operation, rather than individual reads from the socket.
-				m_deadline.expires_from_now(timeout);
-
-				// Set up the variable that receives the result of the asynchronous
-				// operation. The error code is set to would_block to signal that the
-				// operation is incomplete. Asio guarantees that its asynchronous
-				// operations will never fail with would_block, so any other value in
-				// ec indicates completion.
-				//boost::system::error_code ec = boost::asio::error::would_block;
-
-				// Start the asynchronous operation itself. The boost::lambda function
-				// object is used as a callback and will update the ec variable when the
-				// operation completes. The blocking_udp_client.cpp example shows how you
-				// can use boost::bind rather than boost::lambda.
-
-				buff.resize(static_cast<size_t>(sz));
-				boost::system::error_code ec = boost::asio::error::would_block;
-				size_t bytes_transfered = 0;
-
-				
-				handler_obj hndlr(ec, bytes_transfered);
-				async_read((char*)buff.data(), buff.size(), boost::asio::transfer_at_least(buff.size()), hndlr);
-				
-				// Block until the asynchronous operation has completed.
-				while (ec == boost::asio::error::would_block && !m_shutdowned)
-				{
-					m_io_service.run_one(); 
-				}
-
-				if (ec)
-				{
-					LOG_PRINT_L3("Problems at read: " << ec.message());
-          m_connected = false;
-					return false;
-				}else
-				{
-					m_deadline.expires_at(std::chrono::steady_clock::time_point::max());
-				}
-
-				m_bytes_received += bytes_transfered;
-				if(bytes_transfered != buff.size())
-				{
-					LOG_ERROR("Transferred mismatch with transfer_at_least value: m_bytes_transferred=" << bytes_transfered << " at_least value=" << buff.size());
-					return false;
-				}
-
-				return true;
-			}
-
-			catch(const boost::system::system_error& er)
-			{
-				LOG_ERROR("Some problems at read, message: " << er.what());
-        m_connected = false;
-				return false;
-			}
-			catch(...)
-			{
-				LOG_ERROR("Some fatal problems at read.");
-				return false;
-			}
-
```

Comment:
Might be easier to split the dead code piece into its own PR.

Reply:
You're probably right.

Reply:
Done